src/interfaces.ts

```diff
@@ -191,10 +195,12 @@ export interface ICoreKit {
   getCurrentFactorKey(): IFactorKey;
 
   /**
-   * Creates a new factor for authentication.
-   * @param CreateFactorParams - Parameters for creating a new factor.
+   * Creates a new factor for authentication. Generates and returns a new factor
+   * key if no factor key is provided in `params`.
+   * @param params - Parameters for creating a new factor.
+   * @returns The factor key.
    */
-  createFactor(CreateFactorParams: CreateFactorParams): Promise<string>;
+  createFactor(params: CreateFactorParams): Promise<string>;
```

Comment:
for consistency across all functions, it should be `createFactorParams` not `param`

The `C` should be in lowercase.

Reply:
My thinking was: Variable names should be concise. It's clear from context that these are parameters to the function `createFactor`, hence the name `createFactorParams` is redundant.

However, not a big deal. I can change it back for consistency, if you like. Needs to be lower case though.

Reply:
changed as requested